### DISCUSSION TITLE
codegen-16B-multi with 2 T4 each 16GB GPUs error out of memory

### DISCUSSION THREAD

**Discussion by oraclefoo** [moyix/fauxpilot#181] [Q&A]
setup codegen-16B-multi (32GB total VRAM required; multi-language) with 2 T4 gpus. Error with out of memory -
fauxpilot-triton-1         |   what():  [FT][ERROR] CUDA runtime error: out of memory /workspace/build/fastertransformer_backend/build/_deps/repo-ft-src/src/fastertransformer/utils/memory_utils.cu:26
I see the support matrix shows the T4 with 6B, no 16B. And I see the "Note that the VRAM requirements listed by setup.sh are total -- if you have multiple GPUs, you can split the model across them. So, if you have two NVIDIA RTX 3080 GPUs, you should be able to run the 6B model by putting half on each GPU." 

I am expecting the 16B model is split across the 2 T4 GPUs (during setup.sh number of gpus == 2). Launch seems to recognize the two gpus
fauxpilot-triton-1         | I0403 12:39:11.858113 89 pinned_memory_manager.cc:240] Pinned memory pool is created at '0x7ffae4000000' with s
ize 268435456
fauxpilot-triton-1         | I0403 12:39:11.864289 89 cuda_memory_manager.cc:105] CUDA memory pool is created on device 0 with size 67108864
fauxpilot-triton-1         | I0403 12:39:11.864295 89 cuda_memory_manager.cc:105] CUDA memory pool is created on device 1 with size 67108864
fauxpilot-triton-1         | I0403 12:39:12.067798 89 model_repository_manager.cc:1191] loading: fastertransformer:1
fauxpilot-triton-1         | I0403 12:39:12.384258 89 libfastertransformer.cc:1226] TRITONBACKEND_Initialize: fastertransformer

but only created on one GPU
fauxpilot-triton-1         | I0403 12:39:12.388227 89 libfastertransformer.cc:1320] TRITONBACKEND_ModelInstanceInitialize: fastertransformer
_0 (device 0)
fauxpilot-triton-1         | W0403 12:39:12.388240 89 libfastertransformer.cc:453] Faster transformer model instance is created at GPU '0'
fauxpilot-triton-1         | W0403 12:39:12.388242 89 libfastertransformer.cc:459] Model name codegen-16B-multi

Should the 2 T4s work with 16B by splitting the model?

**Comment by dz-star** [comment]
I met the same problem with 2 v100. Have you fix it?
